### DISCUSSION TITLE
dataset = load_dataset("imagefolder", data_dir= data_dir,drop_labels = False)

### DISCUSSION THREAD

**Discussion by cobaltautomationdev** [huggingface/datasets#6249] [Q&A] [answered]
dataset = load_dataset("imagefolder", data_dir= data_dir,**drop_labels = False**), 
I have two image folders  ,
Both almost have same structure, when load imagedata , one image folder need to add **drop_labels = False**, one not. 
if not add **drop_labels = False**, it will not create labels based on dir..
ask bing chat, it says there is a  metadata file present in the directory. but I am sure there is no such a metadata file. 
And I create a new image folder from scratch, it still need to add **drop_labels = False**.

**Comment by mariosasko** [comment]
Can you share the output of running the following code? 
```python
from datasets.data_files import get_metadata_patterns
print(get_metadata_patterns(data_dir))
```

**Reply by cobaltautomationdev** [reply]
without drop_labels = False 
![image](https://github.com/huggingface/datasets/assets/145321686/fa83428f-bc6b-4ddd-bea5-eff5d450f2c1)

metadata file  not exists

![image](https://github.com/huggingface/datasets/assets/145321686/b5ab5ef6-465e-4323-9fb9-1e74aba29845)

**Reply by mariosasko** [reply, marked as answer]
Thanks! I think the problem is that all the files share the same parent directory. If that's the case, there is a single label (equal to the parent directory's name), so we drop it by default (unless `drop_labels=False`).

**Reply by cobaltautomationdev** [reply]
Thanks, I got it.
